### DISCUSSION TITLE
navigateTo not working inside defineNuxtPlugin on initial render

### DISCUSSION THREAD

**Discussion by theathleticnerd** [nuxt/nuxt#27301] [Questions]
I'm trying to navigate the app to a path in my $fetch wrapper plugin. The url to be redirected to is being sent by the API.
If response._data.redirect is true, then it should redirect to response._data.redirect_url_path. 
The code below doesn't work on the initial render (while directly going to the particular url), but it does work after the app loads and the user navigates through the app to the particular url.

This is payload in case of a page that should be redirected:
```javascript
{
    "redirect":true,
    "redirect_url_path":"/xyz/def",
}
``` 

This is the code of the $fetch wrapper.
```typescript
// plugins/customFetch.ts
export default defineNuxtPlugin((nuxtApp) => {
  const config = useRuntimeConfig();
  const $customFetch = $fetch.create({
    baseURL: config.public.baseURL,
    async onRequest({ request, options, error }) {
      options.headers = {
        "X-PUBLIC-URL": `https://www.abcd.com${request}`,
      };
    },
    async onResponse({ response }) {
      if (response._data.redirect) {
        await navigateTo(response._data.redirect_url_path);
      }
    },
  });
  // Expose to useNuxtApp().$customFetch
  return {
    provide: {
      customFetch: $customFetch,
    },
  };
});
``` 
This is the code of the the useFetch wrapper composable
```typescript
// composables/serverFetch.ts
import type { UseFetchOptions } from "nuxt/app";
export function $serverFetch<T>(
  url: string | (() => string),
  options: UseFetchOptions<T> = {},
) {
  return useFetch(url, {
    ...options,
    $fetch: useNuxtApp().$customFetch,
  });
}
``` 
I have tried using nuxtApp.runWithContext and callWithNuxt but it is still not working.
I would greatly appreciate your help.
Thank you.

**Comment by angelhdzmultimedia** [comment]
Try this https://nuxt.com/docs/api/utils/navigate-to#external-url

**Reply by theathleticnerd** [reply]
It isn't an external URL, the path is in the same application. 

I have tried it, but it is not working.
```typescript
 async onResponse({ response }) {
      if (response._data.redirect) {
        await navigateTo(response._data.redirect_url_path, { external: true });
      }
    }
``` 
Thank you for answering

**Reply by angelhdzmultimedia** [reply]
Wish you health, peace, and success. Have fun! 👋😄🔥💚
